internal/fsutil/create.go

```diff
@@ -29,18 +29,9 @@ type Info struct {
 	Link string
 }
 
-type Creator struct {
-	// Created keeps track of information about the filesystem entries created.
-	// If an entry is created several times it only tracks the latest one.
-	Created map[string]Info
-}
-
-func NewCreator() *Creator {
-	return &Creator{Created: make(map[string]Info)}
-}
-
-// Create creates a filesystem entry according to the provided options.
-func (c *Creator) Create(options *CreateOptions) error {
+// Create creates a filesystem entry according to the provided options and returns
+// the information about the created entry.
+func Create(options *CreateOptions) (Info, error) {
```

Comment:
This is a good question. I had assumed that returning a pointer in this case was worse because it will probably incur in a heap allocation that needs to be GC'ed later. Let me see if I can find more information.

Reply:
I have no preference, I only found this bit that might be related in the Go style guide:
> Correctness wins over speed or simplicity. There are cases where you must use a pointer value. In other cases, pick pointers for large types or as future-proofing if you don’t have a good sense of how the code will grow, and use values for simple [plain old data](https://en.wikipedia.org/wiki/Passive_data_structure).

Do you think pointer is better?

Reply:
> pick pointers for large types or as future-proofing if you don’t have a good sense of how the code will grow

I would pick pointers for future-proofing per above.

Reply:
Done.